Feature: add client credentials flow support to KRM

lorenzomasottitn opened the pull request:
This PR enables the OAuth2 client credentials flow by separating audience and clientId in the configuration.

audience continues to represent the resource server(s)

clientId identifies the OAuth2 client application

This allows KRM to support machine-to-machine authentication while maintaining backward compatibility with existing configurations.

Note: A corresponding PR to the digitalhub repository is required to update the KRM Helm chart accordingly.

lorenzomasottitn committed:
Feature: add client credentials flow support to KRM

Introduced client-id property to distinguish between audience and client-id.
Previously, KRM used audience to carry the client identifier, but now the two concepts are handled separately:

audience → represents the intended resource server(s)

client-id → identifies the OAuth2 client application

This change improves clarity and aligns the configuration with standard OAuth2 terminology.

lorenzomasottitn committed:
Feature: add client credentials flow support to KRM

Updated MainController to support the new OAuth2 client credentials flow.
Introduced REACT_APP_CLIENT_ID to distinguish the OAuth2 client ID from the audience.

lorenzomasottitn committed:
Feature: add client credentials flow support to KRM

Added clientId property to AuthenticationProperties to distinguish the OAuth2 client identifier from the audience.
This allows KRM to support the client credentials flow while keeping audience for the resource server.

matteo-s commented on src/main/resources/application.yaml:
```diff
@@ -55,6 +55,7 @@ auth:
   oauth2:
     issuer-uri: ${KRM_AUTH_OAUTH2_ISSUER:}
     audience: ${KRM_AUTH_OAUTH2_AUDIENCE:}
+    client-id: ${KRM_AUTH_OAUTH2_CLIENT_ID:}
```
to be backwards compatible clientId should be configured with `audience` as default value.

lorenzomasottitn replied:
Done

lorenzomasottitn committed:
Add fallback from KRM_AUTH_OAUTH2_CLIENT_ID to KRM_AUTH_OAUTH2_AUDIENCE

Updated application.yaml to support environments where the new
KRM_AUTH_OAUTH2_CLIENT_ID variable is not defined. The client-id
property now falls back to KRM_AUTH_OAUTH2_AUDIENCE if the
primary variable is missing. This ensures compatibility when
deploying Docker images that still rely on the older environment
variables.
